WordPress/src/test/java/org/wordpress/android/ui/mysite/MySiteViewModelTest.kt

```diff
@@ -443,6 +463,42 @@ class MySiteViewModelTest : BaseUnitTest() {
         verify(currentAvatarSource).refresh()
     }
 
+    @Test
+    fun `given mySiteDashboardPhase2FeatureConfig is enabled, when refresh triggers, then all sources are refreshed`() {
+        whenever(mySiteDashboardPhase2FeatureConfig.isEnabled()).thenReturn(true)
+
+        viewModel.refresh()
+
+        allRefreshedMySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach { verify(it).refresh() }
+    }
+
+    @Test
+    fun `given mySiteDashboardPhase2FeatureConfig is disabled, when refresh triggers, then select sources refresh`() {
+        whenever(mySiteDashboardPhase2FeatureConfig.isEnabled()).thenReturn(false)
+
+        viewModel.refresh()
+
+        selectRefreshedMySiteSources.filterIsInstance(MySiteRefreshSource::class.java).forEach { verify(it).refresh() }
```

Comment:
Minor (🔍): Consider also checking for `        verify(selectedSiteRepository).updateSiteSettingsIfNecessary()`, either within this test or a separate test that tests that in isolation.

PS: As per my previous comment, maybe this `selectedSiteRepository.updateSiteSettingsIfNecessary()` should be called even when `mySiteDashboardPhase2FeatureConfig` is enabled. 🤔

Reply:
See previous comment